js/woo.js

```diff
@@ -0,0 +1,1465 @@
+            account['used'] = this.safeString (balance, 'outstanding_holding');
+            account['free'] = Precise.stringAdd (account['total'], account['used']);
+            result[code] = account;
+        }
+        return this.safeBalance (result);
+    }
+
+    async fetchDepositAddress (code, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const [ codeForExchange, networkId ] = this.currencyCodeWithNetwork (currency['code'], params, '_', true);
+        const request = {
+            'token': codeForExchange,
+        };
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchDepositAddress', undefined, params);
+        const method = this.getSupportedMapping (marketType, {
+            'spot': 'v1PrivateGetAssetDeposit',
+        });
+        const response = await this[method] (this.extend (request, query));
+        // {
+        //     success: true,
+        //     address: '3Jmtjx5544T4smrit9Eroe4PCrRkpDeKjP',
+        //     extra: ''
+        // }
+        let tag = this.safeValue (response, 'extra');
+        if (tag === '') {
+            tag = undefined;
+        }
+        const address = this.safeValue (response, 'address');
+        this.checkAddress (address);
+        return {
+            'currency': code,
+            'address': address,
+            'tag': tag,
+            'network': networkId,
+            'info': response,
+        };
+    }
+
+    async fetchDeposits (code = undefined, since = undefined, limit = undefined, params = {}) {
+        const request = {
+            'transaction_type': 'BALANCE',
+            'token_side': 'DEPOSIT',
+        };
+        return await this.fetchTransactions (code, since, limit, this.extend (request, params));
+    }
+
+    async fetchWithdrawals (code = undefined, since = undefined, limit = undefined, params = {}) {
+        const request = {
+            'transaction_type': 'BALANCE',
+            'token_side': 'WITHDRAW',
+        };
+        return await this.fetchTransactions (code, since, limit, this.extend (request, params));
+    }
+
+    async fetchTransactions (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = { };
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+            request['balance_token'] = currency['id'];
+        }
+        if (since !== undefined) {
+            request['start_t'] = since;
+        }
+        if (limit !== undefined) {
+            request['pageSize'] = limit;
+        }
+        const transactionType = this.safeString (params, 'transaction_type');
+        if (transactionType !== undefined) {
+            request['type'] = transactionType;
+        }
+        params = this.omit (params, 'type'); // conflict! we need to change 'type' param for spot/futures into 'exchangeType'
```

Comment:
No conflict here, as with `fetchTansfers` – `fetchTransactions` does not deal with markets, let's cut out these lines.